lib/knapsack_pro/runners/queue/rspec_runner.rb

```diff
@@ -3,224 +3,283 @@
 module KnapsackPro
   module Runners
     module Queue
+      # Imperative Shell [instance of KnapsackPro::Runners::Queue::RSpecRunner]
+      # Architecture based on: https://www.destroyallsoftware.com/talks/boundaries
+      #
+      # It should contain calls to RSpec methods.
+      # Preferably via methods with the  `knapsack__` prefix.
+      # We want to isolate the imperative shell from the internals of RSpec.
+      # It should be tested via integration tests if possible.
+      # Alternatively, it should be tested using E2E tests on CI that are running against a Rails app with the RSpec test suite.
```

Comment:
I'd move the FCore out of this file to make it clear there's a difference of approach.

I'd remove this comment because
- it's more like notes than a code comment
- I don't think we should put youtube links in an OSS gem
- this architecture is still a WIP, so I wouldn't yet state strong boundaries
- we could have it in the PR template as a checklist as a reminder or have a separate internal document for now: core contains pure functions (unit tested), extensions encapsulates calls to RSpec internals (integration and e2e tested), runner invokes the core and the extensions to produce the side-effects (integration and e2e tested)

```suggestion
```

Reply:
I applied changes.